### PR TITLE
feat(core): add validations for export name in CfnOutput

### LINKED ISSUE #28563: (core): CfnOutput does not validate export name during synthesis but can fail during deployment
### Describe the bug

It is possible to create a `CfnOutput` in the CDK that synthesizes without issue, but then fails to deploy due to an invalid export name.

### Expected Behavior

The `CfnOutput` `exportName` property should be validated according to the cloud formation rules at synthesis time to catch this problem sooner in the process.

### Current Behavior

CDK synthesis works just fine, however during deployment Cloud Formation deployments will fail:
```
Deployment failed: Error [ValidationError]: Template format error: Output <SomeOutputNameHere> is malformed. The Name field of every Export member must be specified and consist only of alphanumeric characters, colons, or hyphens.
```

### Reproduction Steps

``` ts
import { CfnOutput, Stack } from 'aws-cdk-lib';

export class MyStack extends Stack {
  constructor(...props: ConstructorParameters<typeof Stack>) {
    super(...props);

    new CfnOutput(this, 'id', {
      value: 'SomeValue',
      exportName: 'SOME_INVALID_EXPORT_NAME`,
    });
  }
}
```

### Possible Solution

Validate the `exportName` field of the `CfnOutput` constructor props during synthesis. Possibly via a regex similar to `[A-Za-z\d\-\:]*`.

### Additional Information/Context

I filed this under bug even though there isn't anything technically "wrong" with the CDK code because as a user, I would expect the synthesis to fail if something that can be validated to conform to cloud formation rules can be validated at synthesis time (and indeed in this case it appears at least on the surface that it can be).

### CDK CLI Version

2.114.1

### Framework Version

_No response_

### Node.js Version

18

### OS

MacOS 14 + Amazon Linux 2 5.10

### Language

TypeScript

### Language Version

5

### Other information

_No response_

### PR DESCRIPTION
**Pull request by go-to-k**
Closes #28563.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by go-to-k** [comment]
Exemption Request: This PR just adds validations, so unit tests could cover them.

File: packages/aws-cdk-lib/core/test/output.test.ts [hunk 1]
```diff
@@ -140,7 +140,44 @@ describe('output', () => {
...
+  test('throw if export name has invalid strings (space)', () => {
+    const output = new CfnOutput(stack, 'SomeOutput', { value: 'x', exportName: 'SOME INVALID EXPORT NAME' });
+    const errors = output.node.validate();
+
+    expect(errors).toEqual([
+      expect.stringContaining('Export name must only include alphanumeric characters, colons, or hyphens (got \'SOME INVALID EXPORT NAME\''),
+    ]);
+  });
+
+  test('throw if export name has invalid strings (under_bar)', () => {
```
Comment (go-to-k) [inline packages/aws-cdk-lib/core/test/output.test.ts:165]:
```suggestion
  test('throw if export name has invalid strings (underscore)', () => {
```

**Review by kylelaker — APPROVED** [review]

**Comment by kylelaker** [comment]
The implementation looks good to me, though I do wonder if this is more of a bugfix along the lines of:

> fix(core): CfnOutput does not validate export name

**Comment by go-to-k** [comment]
@kylelaker 
Thanks for your review!
Recently, I often get the title of PRs for validation additions corrected to `feat`, so I'm using `feat` again. Do you think it should be changed?

https://github.com/aws/aws-cdk/pull/28529
https://github.com/aws/aws-cdk/pull/27592
https://github.com/aws/aws-cdk/pull/27124
https://github.com/aws/aws-cdk/pull/26241

**Comment by kylelaker** [comment]
> @kylelaker Thanks for your review! Recently, I often look to get the title of PRs for validation additions corrected to `feat`, so I'm using `feat` again. Do you think it should be changed?

Oh interesting! Never mind my suggestion then! Thanks for highlighting that; I didn't know that validation additions were being tracked as features. Makes sense to me either way (though I guess it's possibly a little odd that these validation features will basically always need an exemption since there isn't often a README or integration test that makes sense here...) but I won't disagree with the core team when it comes to naming 😄 Thanks for highlighting those previous PRs and I'll try to follow that pattern myself in the future.

Thanks for adding these new validations for `CfnOutput`!

**Comment by gracelu0** [comment]
Thank you for your contribution, we appreciate it! In the future it would be helpful to include a brief description for the PR as well - makes it a bit easier for us to review :)

**Review by gracelu0 — APPROVED** [review]

**Comment by go-to-k** [comment]
@gracelu0 

Thanks for the approval. The merge failed with `Mergify / Queue: Embarked in merge queue`. Could you please handle this again?

**Review by gracelu0 — APPROVED** [review]

**Pull request merged**